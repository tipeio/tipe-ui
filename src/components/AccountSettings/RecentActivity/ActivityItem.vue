<template>
  <div class="item">
    <div class="icon">
      <tipe-icon
        icon="dot"
      />
    </div>
    <div class="text">
      <p class="title">{{ item.title }}</p>
      <p class="time">{{ item.time }}</p>
    </div>

  </div>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '../../Icon'

export default {
  name: 'ActivityItem',
  components: {
    TipeIcon
  },
  props: {
    item: vueTypes.object
  }
}
</script>

<style scoped lang="postcss">
.item {
  display: flex;
  align-items: flex-start;
  width: 100%;
  margin-top: 1rem;
  & .icon {
    display: flex;
    flex: 0 0 1.5rem;
    align-items: flex-start;
    padding: 4px 0 0 4px;
  }

  & .text {
    display: flex;
    flex-direction: column;
    flex: 1 1 auto;

    & .title {
      margin: 0;
      margin-bottom: 7px;
      font-size: 0.75rem;
      color: var(--text-gray);
    }

    & .time {
      margin: 0;
      font-size: 0.625rem;
      color: var(--text-gray-blue-light);
    }
  }
}
</style>
