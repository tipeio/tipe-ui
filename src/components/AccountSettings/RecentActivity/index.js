export { default } from './RecentActivity.vue'
export ActivityItem from './ActivityItem.vue'
