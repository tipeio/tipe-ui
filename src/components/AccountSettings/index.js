export { default } from './AccountSettings.vue'
