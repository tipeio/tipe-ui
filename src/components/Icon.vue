<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
