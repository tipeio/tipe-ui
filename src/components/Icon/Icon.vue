<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
