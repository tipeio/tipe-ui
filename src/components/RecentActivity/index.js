export { default } from './RecentActivity.vue'
