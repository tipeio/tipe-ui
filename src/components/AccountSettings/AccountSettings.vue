<template>
  <div class="content">
    <div class="row">
      <icon-badge icon="file" />
      <space-progress 
        :used-space="8.08" 
        :total-space="40" 
        unit="GB" />
    </div>
    <accordian text="Folder Details"/>
    <accordian text="Settings"><tipe-switch text="File Sharing" /><tipe-switch text="Backup" /></accordian>
    <accordian text="Recent Activity"><activity-item 
      title="Sam Smith update api key" 
      time="a moment ago"/><activity-item 
        title="Sam Smith update api key" 
        time="a moment ago"/></accordian>
  </div>
</template>

<script>
import IconBadge from '../IconBadge'
import Accordian from '../Accordian'
import SpaceProgress from '../SpaceProgress'
import TipeSwitch from '../Switch'
import ActivityItem from '../RecentActivity/ActivityItem'

export default {
  name: 'AccountSettings',
  components: {
    IconBadge,
    Accordian,
    SpaceProgress,
    TipeSwitch,
    ActivityItem
  }
}
</script> 

<style scoped lang="postcss">
.content {
  width: 252px;
  display: grid;
  grid-row-gap: 1.5rem;
}

.row {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-around;
}
</style>
