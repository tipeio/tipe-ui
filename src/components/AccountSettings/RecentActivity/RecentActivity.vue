<template>
  <accordian text="Recent Activity">
    <activity-item 
      v-for="item in items" 
      :key="item.id" 
      :item="item"
    />
  </accordian>
</template>

<script>
import ActivityItem from './ActivityItem'
import Accordian from '../../Accordian'

export default {
  name: 'RecentActivity',
  components: {
    ActivityItem,
    Accordian
  },
  data: function() {
    return {
      items: [
        {
          title: 'Sam Smith update api key',
          time: 'a moment ago',
          id: '47357838'
        },
        {
          title: 'Sarah Scott created a document',
          time: 'one hour ago',
          id: '5389345'
        }
      ]
    }
  }
}
</script>
