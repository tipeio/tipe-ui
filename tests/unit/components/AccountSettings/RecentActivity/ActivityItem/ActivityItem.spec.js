import { shallowMount } from '@vue/test-utils'
import { ActivityItem } from '@/components/AccountSettings/RecentActivity'
import Icon from '@/components/Icon'

describe('ActivityItem.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(ActivityItem, {
      propsData: {
        item: {
          title: 'Olivia broke all of the code',
          subtitle: 'about a minute ago'
        }
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
  it('should contain the item class', () => {
    const wrapper = shallowMount(ActivityItem, {
      propsData: {
        item: {
          title: 'Olivia broke all of the code',
          subtitle: 'about a minute ago'
        }
      }
    })
    expect(wrapper.classes()).toContain('item')
  })
  it('should have the correct item prop', () => {
    const wrapper = shallowMount(ActivityItem, {
      propsData: {
        item: {
          title: 'Olivia broke all of the code',
          subtitle: 'about a minute ago'
        }
      }
    })
    expect(wrapper.props().item).toEqual({
      title: 'Olivia broke all of the code',
      subtitle: 'about a minute ago'
    })
  })
  it('should contain the icon component', () => {
    const wrapper = shallowMount(ActivityItem, {
      propsData: {
        item: {
          title: 'Olivia broke all of the code',
          subtitle: 'about a minute ago'
        }
      }
    })
    expect(wrapper.contains(Icon)).toBe(true)
  })
})
