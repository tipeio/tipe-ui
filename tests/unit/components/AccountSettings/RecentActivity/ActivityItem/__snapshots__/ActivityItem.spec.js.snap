// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityItem.vue renders 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
  "version": 2.5,
}
`;
